src/Utils/OrdersRateLimit.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace AppBundle\Utils;
+
+use AppBundle\Domain\Order\Event;
+use AppBundle\Domain\Order\Event\OrderCreated;
+use AppBundle\Domain\Order\Event\OrderDelayed;
+use AppBundle\Entity\Sylius\Order;
+use Psr\Log\LoggerInterface;
+use Redis;
+
+class OrdersRateLimit
+{
+    public function __construct(
+        private Redis $redis,
+        private LoggerInterface $logger
+    ) {}
+
+
+    /**
+     * @param Order $order
+     * @param \DateTime $pickupTime
+     * @return bool
+     * @throws \RedisException
+     */
+    public function isRangeFull(Order $order, \DateTime $pickupTime): bool {
+
+        if (!$this->featureEnabled($order)) {
+            return false;
+        }
+
+        // Parse order and business parameters
+        $params = $this->getOrdersLimitParameters($order);
+
+        // Calculate time window
+        $start_time = $pickupTime->getTimestamp();
+        $end_time = $start_time + ( $params['max_orders_range_duration'] * 60 );
```

Comment:
Not sure it's the good (or intuitive) way to calculate the time window. 

For example, if we have configured "10 orders per 30 minutes", I would expect to have "stable" time windows, like 10:00-10:30, 10:30:11:00... 

If we calculate the time window like this, it means that if we have a pickup at 10:26, it would generate a time window of 10:26-10:56, i.e a "sliding" time window.

Reply:
After discussing with @r0xsh and @ERAMANrepartos, we agreed that a _sliding_ time window works.